### PR TITLE
[DR-66874] Add 503 and 504 mappings for DR in i18n exception definitions

### PR DESCRIPTION
**Pull request by anniebtran**
## Summary
Whenever we encounter 503s or 504s in our alert triaging, they've always been surfaced as 400 errors in our DataDog dashboards. When we look further into the trace we can see that they're actually 503/4s but we didn't know why we weren't getting those status codes displayed to us in the first place... until we discovered that they were missing a i18n mapping for those codes and the default is to return a 400 if the error code isn't defined in there. So hopefully this will clean up our dashboard view of the errors a little bit and eliminate the extra steps to figure out if a 400 error is actually a 400 or just a 503/4

## Related issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/66874

## Testing done

- [x] *New code is covered by unit tests*

## What areas of the site does it impact?
Decision Reviews backend

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: spec/support/vcr_cassettes/decision_review/HLR-CREATE-RESPONSE-503_V1.yml [hunk 1]
```diff
@@ -0,0 +1,103 @@
+---
```
Comment (anniebtran) [inline spec/support/vcr_cassettes/decision_review/HLR-CREATE-RESPONSE-503_V1.yml:1]: I just copied the 200 response and changed the status to 503 to simulate the error getting raised for the test I added 🤷‍♀️

Comment (anniebtran) [inline spec/support/vcr_cassettes/decision_review/HLR-CREATE-RESPONSE-503_V1.yml:1]: Wasn't sure how to best simulate the 503, so I just copied the 200 response and changed the status, message, and cleaned up the response body to simulate the error getting raised for the test I added 🤷‍♀️ hope that's okay

File: spec/lib/decision_review_v1/service_spec.rb [hunk 2]
```diff
@@ -178,6 +178,19 @@
...
     end
 
+    # test that additional error code mapping in exceptions.en.yml works
+    context '503 response' do
+      it 'properly raises a 503 when the error is returned from the service' do
+        VCR.use_cassette('decision_review/HLR-CREATE-RESPONSE-503_V1') do
+          expect { subject }.to raise_error(
+            an_instance_of(DecisionReviewV1::ServiceException).and(having_attributes(key: 'DR_503')),
+            'BackendServiceException: {:source=>"Common::Client::Errors::ClientError ' \
+            'raised in DecisionReviewV1::Service", :code=>"DR_503"}'
```
Comment (anniebtran) [inline spec/lib/decision_review_v1/service_spec.rb:188]: this code _used_ to be `unmapped_service_exception` but now have proper codes matching the keys in the yaml file

**Review by ericboehs — APPROVED** [review]

**Review by shaunburdick — APPROVED** [review]
Unlike the error messages, this _is_ a timely approval of this PR! LGTM!

**Pull request merged**